crates/chain/src/local_chain.rs

```diff
@@ -354,29 +355,26 @@ impl LocalChain {
     /// the existing chain and invalidate the block after it (if it exists) by including a block at
     /// the same height but with a different hash to explicitly exclude it as a connection point.
     ///
-    /// Additionally, an empty chain can be updated with any chain, and a chain with a single block
-    /// can have it's block invalidated by an update chain with a block at the same height but
-    /// different hash.
+    /// Additionally, a chain with a single block can have it's block invalidated by an update
+    /// chain with a block at the same height but different hash.
     ///
     /// # Errors
     ///
     /// An error will occur if the update does not correctly connect with `self`.
     ///
-    /// Refer to [`Update`] for more about the update struct.
-    ///
     /// [module-level documentation]: crate::local_chain
-    pub fn apply_update(&mut self, update: Update) -> Result<ChangeSet, CannotConnectError> {
-        let changeset = merge_chains(
-            self.tip.clone(),
-            update.tip.clone(),
-            update.introduce_older_blocks,
-        )?;
-        // `._check_index_is_consistent_with_tip` and `._check_changeset_is_applied` is called in
-        // `.apply_changeset`
-        self.apply_changeset(&changeset)
-            .map_err(|_| CannotConnectError {
-                try_include_height: 0,
-            })?;
+    pub fn apply_update(&mut self, update: CheckPoint) -> Result<ChangeSet, CannotConnectError> {
+        let (changeset, can_replace) = merge_chains(self.tip.clone(), update.clone())?;
```

Comment:
I think this would look better if `merge_chains` was implemented on `LocalChain` and took in `&mut self` and `other`. Also if we change `merge_chains` to `merge_and_apply_changeset`, then the business of `can_replace` can stay inside the same function.

idea: evanlinjin/bdk#10

Reply:
I like the idea of returning the result in `merge_chains`. However, removing the `unreachable!` statement doesn't make it nicer imo since we have to rely on `.unwrap`s (also this is just a styling change so I'll just ignore it).

Reply:
agree, the extra refactor is out of scope and more for my own sanity.

Reply:
I've updated to have `merge_chains` return the checkpoint here: 877f7284985c9f1186f902cb4ae24753087eea4d